### DISCUSSION TITLE
Noob question regarding options source schema

### DISCUSSION THREAD

**Discussion by Maduranga** [Budibase/budibase#6405] [Help]
Hello,

How do I find/define the schema for an options picker?
I am looking at pre-defined forms templates and I cannot figure out where the schema data are coming from.
Where are the schemas defined and how can I edit/create new schema?

If there's a reference you can point to, that'd be appreciated.

Cheers.

**Comment by melohagan** [comment]
Hello @Maduranga 

There's a bit on it here: https://docs.budibase.com/docs/option-picker#displaying-options-in-the-option-picker

To try and summarize the three options here:

**Schema**
![Screenshot 2022-06-21 at 10 39 00](https://user-images.githubusercontent.com/101575380/174769061-eef1e0ee-6a80-47c5-b386-90f994ef29b2.png)

**Data Provider**
![Screenshot 2022-06-21 at 10 39 56](https://user-images.githubusercontent.com/101575380/174769302-2041853a-4fae-4b8a-b8a6-c9e7d0522efe.png)

**Custom**
![Screenshot 2022-06-21 at 10 42 17](https://user-images.githubusercontent.com/101575380/174769740-2ccbacf5-1800-416e-b4a7-100465387084.png)
![Screenshot 2022-06-21 at 10 43 02](https://user-images.githubusercontent.com/101575380/174769866-f28448ed-146a-461f-ac6d-2fa84af1a2f7.png)
